src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty()) {
+            throw new AuthenticationException(
+                    "Can't authenticate user by email: " + email);
+        }
+        User user = userByEmail.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Incorrect password");
+    }
```

Comment:
1) It's not good to differentiate error message of wrong email from wrong password in order not to give understanding to potential fraudster if he is correct with some data already
2) Point 1 lead to potential code simplification
```suggestion
        Optional<User> user = userService.findByEmail(email);
        if (user.isPresent() && user.get().getPassword().equals(hashedPassword)) {
            return user.get();
        }
        throw new AuthenticationException("Incorrect login or password");
    }
```